contracts/Liquidator/Liquidator.sol

```diff
@@ -318,12 +392,45 @@ contract Liquidator is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Liqu
         if (!vTokenCollateral.transfer(msg.sender, theirs)) {
             revert VTokenTransferFailed(address(this), msg.sender, theirs);
         }
-        if (!vTokenCollateral.transfer(treasury, ours)) {
-            revert VTokenTransferFailed(address(this), treasury, ours);
+
+        if (IVToken(address(vTokenCollateral)).redeem(ours) != 0) {
+            pendingRedeem.push(address(vTokenCollateral));
+        } else {
+            if (address(vTokenCollateral) == address(vBnb)) {
+                _reduceBnbReserves();
+            } else {
+                _reduceVTokenReserves(address(vTokenCollateral));
+            }
         }
         return (ours, theirs);
     }
 
+    function _reduceBnbReserves() private {
+        uint256 bnbBalance = address(this).balance;
+        IWBNB(wBNB).deposit{ value: bnbBalance }();
+        if (!IWBNB(wBNB).transfer(protocolShareReserve, bnbBalance)) {
+            revert UnderlyingTransferFailed(address(vBnb), wBNB);
+        }
+        IProtocolShareReserve(protocolShareReserve).updateAssetsState(
+            address(comptroller),
+            wBNB,
+            IncomeType.LIQUIDATION
+        );
+        emit ReservesReduced(msg.sender, address(wBNB), bnbBalance);
+    }
+
+    function _reduceVTokenReserves(address vToken) private {
+        address underlying = IVBep20(vToken).underlying();
+        uint256 underlyingBalance = IERC20Upgradeable(underlying).balanceOf((address(this)));
+        IERC20Upgradeable(underlying).safeTransfer(protocolShareReserve, underlyingBalance);
```

Comment:
I would add support to deflationary/rebase tokens, adding a function `doTransferOut` that will return the really transferred amount. So, we can include the real tokens transferred in the event

Reply:
We can keep it as it is. We are only using the amount for the event, and the `updateAssetsState` function is emitting the real transferred amount. So, we can save some gas not doing the extra calculation